vermouth/tests/test_apply_rubber_band.py

```diff
@@ -0,0 +1,126 @@
+            node['chain'] = 'B'
+    return graph
+
+
+@pytest.mark.parametrize('separation', (0, 1, 2, 3, 7))
+@pytest.mark.parametrize('selection', (
+    None,  # Use all the nodes, implicitly
+    list(range(16)),  # Use all the nodes, explicitly
+    list(range(8)),  # Only the first component
+    list(range(8, 16)),  # Only the second component
+    list(range(0, 16, 2)),  # Every other nodes
+))
+def test_build_connectivity_matrix(disconnected_graph, separation, selection):
+    connectivity = apply_rubber_band.build_connectivity_matrix(
+        disconnected_graph, separation, selection)
+    other_component = [np.inf] * 8
+    distance = np.array([
+        #0  1  2  3  4  5  6  7    8 to 15
+        [0, 0, 1, 2, 6, 5, 4, 3] + other_component,  # 0
+        [0, 0, 0, 1, 5, 4, 3, 2] + other_component,  # 1
+        [1, 0, 0, 0, 4, 3, 2, 1] + other_component,  # 2
+        [2, 1, 0, 0, 3, 2, 1, 0] + other_component,  # 3
+        [6, 5, 4, 3, 0, 0, 1, 2] + other_component,  # 4
+        [5, 4, 3, 2, 0, 0, 0, 1] + other_component,  # 5
+        [4, 3, 2, 1, 1, 0, 0, 0] + other_component,  # 6
+        [3, 2, 1, 0, 2, 1, 0, 0] + other_component,  # 7
+        # 0 to 7           8  9 10 11 12 13 14 15
+        other_component + [0, 0, 1, 2, 0, 1, 2, 3],  # 8
+        other_component + [0, 0, 0, 1, 1, 2, 1, 2],  # 9
+        other_component + [1, 0, 0, 0, 2, 1, 0, 1],  # 10
+        other_component + [2, 1, 0, 0, 3, 2, 1, 0],  # 11
+        other_component + [0, 1, 2, 3, 0, 0, 1, 2],  # 12
+        other_component + [1, 2, 1, 2, 0, 0, 0, 1],  # 13
+        other_component + [2, 1, 0, 1, 1, 0, 0, 0],  # 14
+        other_component + [3, 2, 1, 0, 2, 1, 0, 0],  # 15
+    ])
+    np.fill_diagonal(distance, np.inf)
+    expected_connectivity = (distance <= separation)
+    if selection is not None:
+        expected_connectivity = expected_connectivity[:, selection][selection]
+    assert np.all(connectivity == expected_connectivity)
+
+
+@pytest.mark.parametrize('selection', (
+        None,  # Use all the nodes, implicitly
+        list(range(16)),  # Use all the nodes, explicitly
+        list(range(8)),  # Only the first component
+        list(range(8, 16)),  # Only the second component
+        list(range(0, 16, 2)),  # Every other nodes
+))
+@pytest.mark.parametrize('extra_edges', ([], [(7, 10)]))
+def test_build_pair_matrix(disconnected_graph, selection, extra_edges):
+    """
+    The creation of a pair matrix works as expected.
+
+    Here, the criterion is ``True`` when nodes belong to the same domain.
+    The graph is defined as having two domains: one per chain. The extra edges
+    allow to make sure the connectivity does not impact the domain detection.
+    """
+    def have_same_chain(graph, left, right):
+        return graph.nodes[left]['chain'] == graph.nodes[right]['chain']
+
+    expected = np.zeros((16, 16), dtype=bool)
+    expected[:8, :8] = True
+    expected[8:, 8:] = True
+    np.fill_diagonal(expected, False)
```

Comment:
Should this be needed?

Reply:
@pckroon In what sense is this not needed?

Reply:
It is needed indeed. We don't expect rubber bands between a node and itself.